libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -760,6 +759,7 @@ void DeclarationRegistrationHelper::registerDeclaration(Declaration& _declaratio
 	registerDeclaration(*m_scopes[m_currentScope], _declaration, nullptr, nullptr, warnAboutShadowing, inactive, m_errorReporter);
 
 	_declaration.annotation().scope = m_currentScope;
+	_declaration.annotation().contract = m_currentContract;
```

Comment:
To be honest, I think we should add a function to the annotation (or a utility function) to determine the contract from the scope instead of storing it.

Reply:
Maybe... it's not like I added it - I just moved it from the annotations of function definitions to declarations - which wouldn't even have been necessary for this PR, but *if* we have the annotation, then this is the better place for it...